client/util.go

```diff
@@ -21,17 +22,15 @@ const docIDPrefixLength = docIDVersionLength + docIDNonceLength
 // pathnameToDocID encrypts a `pathname` to a document ID using `key`.
 func pathnameToDocID(pathname string, key [32]byte) (sserver1.DocumentID, error) {
 	var nonce [docIDNonceLength]byte
-	_, err := rand.Read(nonce[:])
-	if err != nil {
-		return sserver1.DocumentID(""), err
-	}
+	cksum := sha256.Sum256([]byte(pathname))
```

Comment:
I guess it needs to be deterministic so that renames and deletes will work?  (Searches would work fine with a random nonce since the clients would just learn the nonce and docID together.)  Can you please add a comment here about that?  Deterministic nonces seems a little scary to me, but maybe there's a reason it's ok here.